src/Brancho/Resolver/JiraResolver.php

```diff
@@ -45,46 +45,55 @@ class JiraResolver extends AbstractResolver implements ConfigurableResolverInter
      */
     public function resolve(InputInterface $input, OutputInterface $output, ContextInterface $context): ?array
     {
-        $issue = $this->getIssue($input, $output);
+        $issueKey = $this->getIssue($input, $output);
 
         $filter = $context->getFilter();
         $config = $context->getConfig()['jira'];
 
-        $jiraIssue = $this->getFactory()->createJira()->getJiraIssue($issue, $config);
+        $jiraIssueData = $this->getFactory()->createJira()->getJiraIssue($issueKey, $config);
 
-        if (isset($jiraIssue['errorMessages'])) {
-            foreach ($jiraIssue['errorMessages'] as $errorMessage) {
+        if (isset($jiraIssueData['errorMessages'])) {
+            foreach ($jiraIssueData['errorMessages'] as $errorMessage) {
                 $output->writeln(sprintf('<fg=red>%s</>', $errorMessage));
             }
 
             return null;
         }
 
-        $issue = $filter->filter($issue);
-        $issueType = $filter->filter($jiraIssue['fields']['issuetype']['name']);
-        $summary = $filter->filter($jiraIssue['fields']['summary']);
+        $issueKey = $filter->filter($issueKey);
+        $issueType = $filter->filter($jiraIssueData['fields']['issuetype']['name']);
+        $issueSummary = $filter->filter($jiraIssueData['fields']['summary']);
 
         if ($issueType === 'bug') {
-            return (array)$this->createBugfixBranchName($issue, $summary);
+            return (array)$this->createBugfixBranchName($issueKey, $issueSummary);
         }
 
         if ($issueType === 'epic') {
-            return $this->createEpicBranchNames($input, $output, $issue, $summary);
+            return $this->createEpicBranchNames($input, $output, $issueKey, $issueSummary);
         }
 
-        $epicOrStoryJiraIssue = $this->getParentJiraIssue($jiraIssue, $config);
-        $epicOrStoryIssue = $filter->filter($epicOrStoryJiraIssue['key']);
-        $epicOrStoryIssueType = $filter->filter($epicOrStoryJiraIssue['fields']['issuetype']['name']);
-        $issue = sprintf('%s/%s', $epicOrStoryIssue, $issue);
+        $finalIssueSlug = $issueKey;
```

Comment:
Why are we moving the `issueKey` to `finalIssueSlug` instead of re-assigning the `issueKey` if we need to?
Also, a slug and issue key are two different things. The `createBranchName` function's first argument is `issue`, but `slug` is being passed, which is weird.

Reply:
I am sorry about changing the variables, it was really confusing to understand what was going on.. For example, I had a variable called `$issue` and while I was reading the code I thought it was the issue entity itself (with its data) but in fact, it was only the issue key.. so, because of that and other things, was really difficult to follow the code, renaming them made easy for me to understand.

but regarding what you mentioned that's ok, I fixed it